packages/internal/src/utils.js

```diff
@@ -250,16 +250,12 @@ harden(bindAllMethods);
 
 /**
  * @template {{}} T
- * @typedef {{ [K in keyof T]: DeeplyAwaited<T[K]> }} DeeplyAwaitedObject
+ * @typedef {{ [K in keyof T]: T[K] extends Function ? T[K] : DeeplyAwaited<T[K]> }} DeeplyAwaitedObject
  */
 
 /**
- * Caveats:
- * - doesn't recur within Promise results
- * - resulting type has wrapper in its name
- *
  * @template T
- * @typedef {T extends PromiseLike<any> ? Awaited<T> : T extends {} ? DeeplyAwaitedObject<T> : Awaited<T>} DeeplyAwaited
+ * @typedef {T extends PromiseLike<any> ? Awaited<T> : T extends {} ? import('type-fest').Simplify<DeeplyAwaitedObject<T>> : Awaited<T>} DeeplyAwaited
```

Comment:
Is this the first time we take an external dependency for our types? Wondering how that works out once we start publishing our packages correctly and the exported types have external dependencies (there is an argument to be made in that case that it is not in fact a dev dependency)

Reply:
Good catch! `yarn prepack` showed this change didn't even build.

It's such a small type I backed out the import and copied it over with attribution.